app/Controller/AppController.php

```diff
@@ -234,7 +234,9 @@ public function beforeFilter()
                 if ($loginByAuthKeyResult === false || $this->Auth->user() === null) {
                     if ($loginByAuthKeyResult === null) {
                         $this->loadModel('Log');
-                        $this->Log->createLogEntry('SYSTEM', 'auth_fail', 'User', 0, "Failed API authentication. No authkey was provided.");
+                        $this->UserLoginProfile = ClassRegistry::init('UserLoginProfile');
+                        $change = $this->UserLoginProfile->_getUserProfile();
```

Comment:
Excellent remark and question. The goal here was to log more details of the browser and so, in the same structured way. However considering the slow function, and this exact section is when no auth key was provided, it could facilitate a DoS resource exhaustion attack. 
I will remove this section.